File: back-end/ConstructionFinance/src/main/java/br/com/galsystem/construction/finance/models/Expense.java
package br.com.galsystem.construction.finance.models;
import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "expense")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Expense {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id")
    private User user;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false)
    private String description;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne
    @JoinColumn(name = "supplier_id")
    private Supplier supplier;

    @ManyToOne
    @JoinColumn(name = "payer_id")
    private Payer payer;

    private String paymentMethod;

    @Column(nullable = false)
    private BigDecimal amount;

    private String attachmentUrl;
}

File: front-end/src/app/pages/expense/expense.ts
import { Payer } from "../payer/Payer";
import { ServiceContractDTO } from "../serviceContract/service-contract.dto";
import { Supplier } from "../supplier/supplier";

export interface Expense {
  id: number;
  date: string;           
  description: string;
  supplierId: number;
  payerId: number;
  serviceContractId: number;
  paymentMethod: string;
  amount: number;        
  attachmentUrl: string | null;
  supplier: Supplier;
  payer: Payer;
  serviceContract: ServiceContractDTO;
}